SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1441 @@
+	GetEnvironmentPasswordPolicyDetail -ppmVariables $ppmVariables -ppmEnvironmentalDetails $ppmEnvironmentalDetails
+	$count = 0
+		#addressing comparison reports
+	if ($doComparison -eq $true) {
+		foreach ($eachNode in $ppmEnvironmentalDetails) {
+			foreach ($eachAttribute in $eachNode.productAttributes.PSObject.Properties){
+				foreach ($product in $ppmStandardConfigValues) {
+					if ($product.productType -eq $eachNode.productType) {
+						if ($eachAttribute.Value -ne $null) {
+							if ($eachAttribute.Value -ne $product.productAttributes.$($eachAttribute.Name)) {
+								$eachAttribute.Value = "<font color=#DC143C><b>" + $eachAttribute.Value + "</b></font>["+$product.productAttributes.$($eachAttribute.Name)+"]"
+								$count += 1
+							}
+						} else{
+							$eachAttribute.Value = "<font color=#DC143C><b>NULL</b></font>["+$product.productAttributes.$($eachAttribute.Name)+"]"
+							$count += 1
+						}
+					}
+				}
+			}
+		}
+	}
+	#prepare html header
+	$htmlOutput = "<html><title>Password Policy Report</title><style type=" + '"text/css"' + ">.myTable { border-collapse:collapse; }.myTable td, .myTable th {text-align:center;padding:5px;border:1px solid #000;}</style><head><H1>Password Policy Report for the Workload domain <font color=#6495ED>$($ppmVariables.sddcDomainName)</font> </H1></head><body><br><H4><p>-Report Ran on <font color=#6495ED>" + (Get-Date).tostring("dd-MM-yyyy-hh-mm-ss") + "</font> for the Workload domain <font color=#6495ED>$($ppmVariables.sddcDomainName)</font>"
+	if ($doComparison -eq $false) {
+		$htmlOutput = $htmlOutput + "<br><font color=#FFA500>-Common password policy file not provided, Comparison has not been evaluated</font></p></H4>"
+	} else {
+		$htmlOutput = $htmlOutput + "</p></H4>"
+	}
+	#prepare ESXi servers section"
+	$htmlOutput = $htmlOutput + '<H3>ESXi servers Password Policy</h3><table class="myTable">'
+	$tmpHtmlOutput = $ppmEnvironmentalDetails | where { $_.productType -match "ESXi" } | select-object -Property FQDN -ExpandProperty productAttributes | select-object FQDN,passwdExpInDays,passwdMaxFailAttempts,passwdMinimumLengthFor1CharClass,passwdMinimumLengthFor2CharClass,passwdMinimumLengthFor3CharClass,passwdMinimumLengthFor4CharClass,passwdMinimumCharLengthForPhrase | sort-object -Property FQDN | ConvertTo-Html -Fragment -As Table
+	$htmlOutput += $tmpHtmlOutput
+	#prepare vCenter Servers section"
+	$htmlOutput = $htmlOutput + '<H3>vCenter Server Password Policy</h3><table class="myTable">'
+	$tmpHtmlOutput = $ppmEnvironmentalDetails | where { $_.productType -match "VC" } | select-object -Property FQDN -ExpandProperty productAttributes | select-object FQDN,passwdExpInDays,passwdNotifyEmail | sort-object -Property FQDN | ConvertTo-Html -Fragment -As Table
+	$htmlOutput += $tmpHtmlOutput
+	#prepare SSO section"
+	$htmlOutput = $htmlOutput + '<H3>Single Sign On Password Policy</H3><p><font color=#6495ED>SSO FQDN: ' + $ppmVariables.ssoServerFqdn + '</font></p><table class="myTable">'
```

Comment:
"vCenter Single Sign-On Password Policy"

Reply:
addressed in 99029d5ec8015e0c5c645e9abc7c3384e1b3091e